bundles/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/PulseAudioAudioSource.java

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.pulseaudio.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.Socket;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.pulseaudio.internal.handler.PulseaudioHandler;
+import org.openhab.core.audio.AudioException;
+import org.openhab.core.audio.AudioFormat;
+import org.openhab.core.audio.AudioSource;
+import org.openhab.core.audio.AudioStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The audio source for openhab, implemented by a connection to a pulseaudio source using Simple TCP protocol
+ *
+ * @author Miguel Álvarez - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class PulseAudioAudioSource extends PulseaudioSimpleProtocolStream implements AudioSource {
+
+    private final Logger logger = LoggerFactory.getLogger(PulseAudioAudioSource.class);
+
+    private HashSet<AudioFormat> supportedFormats = new HashSet<>();
+
+    public PulseAudioAudioSource(PulseaudioHandler pulseaudioHandler, ScheduledExecutorService scheduler) {
+        super(pulseaudioHandler, scheduler);
+        var audioFormat = pulseaudioHandler.getSourceAudioFormat();
+        if (audioFormat != null) {
+            supportedFormats.add(audioFormat);
+        }
+    }
+
+    @Override
+    public Set<AudioFormat> getSupportedFormats() {
+        return supportedFormats;
+    }
+
+    @Override
+    public AudioStream getInputStream(AudioFormat audioFormat) throws AudioException {
+        try {
+            for (int countAttempt = 1; countAttempt <= 2; countAttempt++) { // two attempts allowed
+                try {
+                    connectIfNeeded();
+                    final Socket clientSocketLocal = clientSocket;
+                    if (clientSocketLocal == null) {
+                        break;
+                    }
+                    var sourceFormat = pulseaudioHandler.getSourceAudioFormat();
+                    if (sourceFormat == null) {
+                        throw new AudioException("Unable to get source audio format");
+                    }
+                    if (!audioFormat.isCompatible(sourceFormat)) {
+                        throw new AudioException("Incompatible audio format requested");
+                    }
+                    setIdle(true);
```

Comment:
I have a doubt, is it true or false you want to set here ?

Reply:
True, so the underling socket will be closed if the stream remains unused, that's ok as it will reconnect on first read.